src/bp/ui-shared/src/telemetry/offlineServerFallback.ts

```diff
@@ -0,0 +1,49 @@
+import axios, { AxiosInstance } from 'axios'
+import { TelemetryEvent } from 'common/telemetry'
+import ms from 'ms'
+
+export const sendEventsToTelemetryServer = async (events: TelemetryEvent[]) => {
+  try {
+    await axios.post(
+      window.TELEMETRY_URL,
+      events.map(e => ({ ...e, source: 'client' }))
+    )
+    return true
+  } catch (err) {
+    console.error('Could not send the telemetry packages to the storage server', err)
+    return false
+  }
+}
+
+export const startOfflineTelemetryFallback = (api: AxiosInstance) => {
+  sendSavedEventsRecursive()
+    .then(async () => {
+      setInterval(async () => await sendSavedEventsRecursive(), ms('30m'))
+    })
+    .catch() // silently fail
```

Comment:
Please use async/await with try/catch. We avoid then/catch when possible. Code is easier to read

Reply:
I could use try/catch, but tslint would always display a warning saying that "Promise should be handled appropriately". Which is why this syntax was used. It removed this warning

Reply:
`try {
    sendSavedEvents()
    setInterval(async () => await sendSavedEvents(), ms('30m'))
  } catch (err) {}`
the "sendSavedEvents()" gets the tslint warning

Reply:
```suggestion
    try {
    sendSavedEvents()
    setInterval(async () => await sendSavedEvents(), ms('30m'))
  } catch (err) {}
```
the "sendSavedEvents()" has a tslint warning

Reply:
What about `await sendSavedEvents()` ?

Reply:
Fixed in 7e3743e!